routes/apiLists.js

```diff
@@ -0,0 +1,24 @@
+var express = require('express');
+var router = express.Router();
+var List = require('../models/list');
+
+var seedDB = require('../seed/lists');
+seedDB();
+
+// get a list of lists from the db
+router.get('/list', function(req, res, next){
+  List.find({})
+    .populate(
+      {
```

Comment:
Can you explain what the thing in populate is?

Reply:
why is the path cards ?
I have not seen anything in the populate upto now just populate()

Reply:
If you look at the list schema you will see the following as a field:
```
cards: [
    {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Card"
    }
]
```
When a card is created in the list, it will create a card in the cards collection.
Then you want to link the card with its list, to do so you push the card to the cards array in the list Schema. With the code above only the ID of the card is pushed, instead of the whole card Schema.

The populate method allows to query for the card data and put it in the response, if you don't you will only get the ID and not the content of the card.

The path "cards" refers to the field in the List schema. There is another populate nested inside because I need to populate with the comments too.